### PR TITLE
go_1_19: init at 1.19

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

https://go.dev/doc/go1.19

### CODE REVIEW

**Comment by zowoq** [comment]
`creds-test.patch` was upstreamed.

```diff 
 stdenv.mkDerivation rec {
   pname = "go";
-  version = "1.18.4";
+  version = "1.19rc2";
 
   src = fetchurl {
     url = "https://go.dev/dl/go${version}.src.tar.gz";
-    sha256 = "sha256-RSWqaw487LV4RfQGCnB1qvyat1K7e2tM+KIS1DB44eQ=";
+    sha256 = "sha256-xo1wGaaguYUq4Oltfn3rdySSojJy/W0Tr+BbQMkS5Rs=";
   };
 
   strictDeps = true;
@@ -176,11 +176,10 @@
     ./remove-tools-1.11.patch
     ./ssl-cert-file-1.16.patch
     ./remove-test-pie-1.15.patch
-    ./creds-test.patch
     ./skip-chown-tests-1.16.patch
     ./skip-external-network-tests-1.16.patch
     ./skip-nohup-tests.patch
-    ./skip-cgo-tests-1.15.patch
+    ./skip-cgo-tests-1.19.patch
     ./go_no_vendor_checks-1.16.patch
   ];
 

```

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -21970,6 +21970,17 @@ with pkgs;
...
+  # requires a newer Apple SDK
+  go_1_19 = darwin.apple_sdk_11_0.callPackage ../development/compilers/go/1.19.nix {
+    inherit (darwin.apple_sdk_11_0.frameworks) Foundation Security;
+  };
+  buildGo119Module = darwin.apple_sdk_11_0.callPackage ../development/go-modules/generic {
+    go = buildPackages.go_1_19;
+  };
+  buildGo119Package = darwin.apple_sdk_11_0.callPackage ../development/go-packages/generic {
+    go = buildPackages.go_1_19;
+  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:21982]: Do we really need thi?

**Review by SuperSandro2000 — COMMENTED** [review]
I think we should take this opportunity to convert the file into a generic one. Copying it with every new go release just duplicates code.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (zowoq) [reply pkgs/top-level/all-packages.nix:21982]: > Do we really need thi?

If you are asking if we need `buildGo119Package`, as long as we are still using `buildGoPackage` we need a versioned equivalent.

**Comment by zowoq** [comment]
> I think we should take this opportunity to convert the file into a generic one. Copying it with every new go release just duplicates code.

This has been discussed before and the `go` maintainers were fine with the status quo. Even if we do decide to switch to a generic style derivation it doesn't need to happen now.

**Comment by zimbatm** [comment]
Result of `nixpkgs-review pr 181797` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>go_1_19</li>
  </ul>
</details>

**Comment by Mic92** [comment]
> > I think we should take this opportunity to convert the file into a generic one. Copying it with every new go release just duplicates code.
> 
> This has been discussed before and the `go` maintainers were fine with the status quo. Even if we do decide to switch to a generic style derivation it doesn't need to happen now.

Yeah with 2-3 go derivations is just simply easier to only have too look at one file rather than a generic function and one per go version.

**Comment by xvello** [comment]
Thanks for the packaging work @zowoq! Do you know whether this will be backported in 22.05 or not?

**Comment by Mic92** [comment]
@xvello I don't see why not. Feel to create a backport PR.

**Comment by xvello** [comment]
Thanks, I'll look into how to do so.

**Pull request merged**